### DISCUSSION TITLE
interpolateStyles

### DISCUSSION THREAD

**Discussion by dillingham** [remotion-dev/remotion#3289] [Ideas]
Had an idea Im unable to work on but wanted to share regardless

### interpolateStyles

Like a CSS transition in defining the two css states but letting remotion figure out the in between.

#### Example usage

```js
const from = {
    width: 100,
}

const to = {
    width: 200,
}

const styles = interpolateStyles(from, to, 20)

return <div style={styles}>hello</div>
```

#### Simple proof of concept

```js
import { interpolate, useCurrentFrame } from "remotion";

const interpolateStyles = (from, to, duration) => {
  const frame = useCurrentFrame()

  let styles = {}
  
  Object.keys(from).forEach(key => {
    styles[key] = interpolate(frame, [0, duration], [from[key], to[key]])
  })

  return styles;
}

export default interpolateStyles;
```
### Cases

- if color property, use interpolateColor()
- if numeric, use interpolate(), append the unit px 
- if transform, break down string and use interpolate
- maybe skip any keys that exist in `from` but not in `to`
- background: 'red' would be hard because can be many things, maybe only backgroundColor

**Comment by JonnyBurger** [comment]
I like this idea! 💡 Thanks for sharing!
